lib/zold/node/entrance.rb

```diff
@@ -37,7 +37,7 @@ module Zold
   class Entrance
     def initialize(wallets, remotes, copies, address, log: Log::Quiet.new)
       raise 'Wallets can\'t be nil' if wallets.nil?
-      raise 'Wallets must be of type Wallets' unless wallets.is_a?(Wallets)
+      raise 'Wallets must quack like Wallets' unless wallets.respond_to?(:find)
```

Comment:
I believe, it should just check class. Whether it responds to `#find` should be checked in tests. Wallets support other methods as well, and it would be a downgrade in product quality to check just for one method.

Reply:
My reasoning is: there are many methods of `Wallets`, you must either check for all methods that are used in the code or you won't get the error message. It is also error prone - there is high chance, the checks won't be updated. I think it's a bad system to check class, but changing it is out of scope.

So, if you added check to `HungryWallets`, that it gets passed `Wallets`, that would leave the existing functionality intact. And checking `is_a?(HungryWallets)` would continue the same way of doing things.

Reply:
@ledestin fair enough, makes sense.